src/librustc/ich/impls_hir.rs

```diff
@@ -393,11 +393,12 @@ impl<'a> HashStable<StableHashingContext<'a>> for hir::TraitCandidate {
         hcx.with_node_id_hashing_mode(NodeIdHashingMode::HashDefPath, |hcx| {
             let hir::TraitCandidate {
                 def_id,
-                import_id,
-            } = *self;
+                import_ids,
+            } = self;
 
             def_id.hash_stable(hcx, hasher);
-            import_id.hash_stable(hcx, hasher);
+            // We only use the outermost import NodeId as key
```

Comment:
`import_ids` are only ever used as a whole set, so the first import is not special and the whole `import_ids` needs to be hashed (here and below).

Reply:
Fixed in 4188df7